docs/examples/userguide/numpy_tutorial/compute_prange.pyx

```diff
@@ -16,7 +14,7 @@ ctypedef fused my_type:
 
 
 # We declare our plain c function nogil
-cdef my_type clip(my_type a, my_type min_value, my_type max_value) nogil:
+cdef my_type clip(my_type a, my_type min_value, my_type max_value) noexcept nogil:
```

Comment:
I think we should explain why we need `noexcept` here...

Reply:
Is this just that exception values for fused types are currently handled badly?

Reply:
No this is because `clip` function is called with GIL released (caused by `prange(nogil=True)`). Without `noexcept` performance warning is raised.

Reply:
So I suppose the documentation should said something like:

> `noexcept` keyword is needed to avoid performance overhead caused by GIL acquiring during  exception check

or something like that. I need to figure out correct wording.

Reply:
That's actually what I meant. It should be possible for Cython to use `-1` as the error value, but it doesn't.

https://github.com/cython/cython/pull/5737 should fix that. But it may not be immediate

Reply:
I have added a note with the link to the issue.